File: library/python/libcarlab/registry.py

from typing import *

class Information:
    def __init__(self, name, datatype):
        self.name = name
        self.datatype = datatype

    def __hash__(self):
        return hash(self.name)

    def __eq__(self, other):
        return (
            self.__class__ == other.__class__ and
            self.name == other.name
        )


# Registry
class Registry:
    MapMatchedLocation = Information('map-matched-location', Tuple[str,float])
    Location = Information('location', Tuple[float, float])
    CarSpeed = Information('car-speed', float)
    CarGear = Information('car-gear', int)
    CarFuel = Information('car-fuel', float)
    CarSteering = Information('car-steering', float)
    VehiclePointingRotation = Information('vehicle-pointing-rotation', List[float])
    WorldPointingRotation = Information('world-pointing-rotation', List[float])
    VehicleAlignedAccel = Information('vehicle-aligned-accel', Tuple[float, float, float])
    WorldAlignedAccel = Information('world-aligned-accel', Tuple[float, float, float])
    WorldAlignedGyro = Information('world-aligned-gyro', Tuple[float, float, float])
    GravityAlignedGyro = Information('gravity-aligned-gyro', float)
    GearModelFile = Information('gear-model-file', str)
    CarModel = Information('car-model', str)
    PhoneNumber = Information('phone-number', str)
    UserText = Information('user-text', str)
    Sighting = Information('sighting', Tuple[float, float, float])
    SightingsMap = Information('sightings-map', List[List[float]])
    GPS = Information('gps', Tuple[float, float, float])
    Accel = Information('accel', Tuple[float, float, float])
    Magnetometer = Information('magnetometer', Tuple[float, float, float])
    Gyro = Information('gyro', Tuple[float, float, float])
    Gravity = Information('gravity', Tuple[float, float, float])
    ObdFuel = Information('obd-fuel', float)

File: library/react/libcarlab/src/registry.ts

export class Information {
    name: string;
    dtype: any;
  
    constructor(name: string, dtype: any) {
      this.name = name;
      this.dtype = dtype;
    }
  }
  
  export class Registry {
      static MapMatchedLocation = new Information('map-matched-location', 0.0)
      static Location = new Information('location', 0.0)
      static CarSpeed = new Information('car-speed', 0.0)
      static CarGear = new Information('car-gear', 0.0)
      static CarFuel = new Information('car-fuel', 0.0)
      static CarSteering = new Information('car-steering', 0.0)
      static VehiclePointingRotation = new Information('vehicle-pointing-rotation', 0.0)
      static WorldPointingRotation = new Information('world-pointing-rotation', 0.0)
      static VehicleAlignedAccel = new Information('vehicle-aligned-accel', 0.0)
      static WorldAlignedAccel = new Information('world-aligned-accel', 0.0)
      static WorldAlignedGyro = new Information('world-aligned-gyro', 0.0)
      static GravityAlignedGyro = new Information('gravity-aligned-gyro', 0.0)
      static GearModelFile = new Information('gear-model-file', 0.0)
      static CarModel = new Information('car-model', 0.0)
      static PhoneNumber = new Information('phone-number', 0.0)
      static UserText = new Information('user-text', 0.0)
      static Sighting = new Information('sighting', 0.0)
      static SightingsMap = new Information('sightings-map', 0.0)
      static GPS = new Information('gps', 0.0)
      static Accel = new Information('accel', 0.0)
      static Magnetometer = new Information('magnetometer', 0.0)
      static Gyro = new Information('gyro', 0.0)
      static Gravity = new Information('gravity', 0.0)
      static ObdFuel = new Information('obd-fuel', 0.0)
  }
  
  
  